bfabric/bfabric_config.py

```diff
@@ -27,73 +30,146 @@ class BfabricConfig:
             application_ids=application_ids
             if application_ids is not None
             else self.application_ids,
         )
 
 
-def parse_bfabricrc_py(file: io.FileIO) -> Tuple[BfabricConfig, Optional[BfabricAuth]]:
+'''
+NOTE: BFabricPy expects a .bfabricpy.yml of the format, as seen in bfabricPy/tests/unit/example_config.yml
+* The general field always has to be present
+* There may be any number of environments, and they may have arbitrary names. Here, they are called PRODUCTION and TEST
+* Must specify correct login, password and webbase for each environment.
+* application and job_notification_emails fields are optional
+* The default environment will be selected as follows:
+    - First, parser will check if the optional argument `config_env` is provided directly to the parser function
+    - If not, secondly, the parser will check if the environment variable `BFABRICPY_CONFIG_ENV` is declared
+    - If not, finally, the parser will select the default_config specified in [GENERAL] of the .bfabricpy.yml file 
+'''
+
+
+def _read_config_env_as_dict(config_path: str, config_env: str = None) -> Tuple[str, dict]:
+    """
+    Reads and partially parses a bfabricpy.yml file
+    :param config_path:  Path to the configuration file. It is assumed that it exists
+    :param config_env:   Specific environment to parse. If not provided, it is deduced from an environment variable
+       or the config file itself.
+    :return: Returns a target environment name, and the corresponding data from bfabricpy.yml file as a dictionary
+    """
+
     """Parses a .bfabricrc.py file and returns a tuple of BfabricConfig and BfabricAuth objects."""
-    values = {}
-    file_path = os.path.realpath(file.name)
     logger = logging.getLogger(__name__)
-    logger.info(f"Reading configuration from: {file_path}")
-
-    for line in file:
-        if line.startswith("#"):
-            continue
-
-        key, _, value = [part.strip() for part in line.partition("=")]
-        if key not in [
-            "_PASSWD",
-            "_LOGIN",
-            "_WEBBASE",
-            "_APPLICATION",
-            "_JOB_NOTIFICATION_EMAILS",
-        ]:
-            continue
-
-        # In case of multiple definitions, the first rule counts!
-        if key not in values:
-            if key in ["_APPLICATION"]:
-                try:
-                    values[key] = json.loads(value)
-                except json.JSONDecodeError as e:
-                    raise ValueError(
-                        f"While reading {file_path}. '{key}' is not a valid JSON string."
-                    ) from e
-            else:
-                # to make it downward compatible; so we replace quotes in login and password
-                values[key] = value.replace('"', "").replace("'", "")
+    logger.info(f"Reading configuration from: {config_path}")
+
+    # Read the config file
+    config_dict = yaml.safe_load(Path(config_path).read_text())
```

Comment:
Somehow, I didn't know until this day, that there is `Path.read_text`, thanks a lot!

Reply:
Found it on stackoverflow.